chromium_src/chrome/browser/ui/browser_commands.cc

```diff
@@ -5,14 +5,48 @@
 
 #include "brave/browser/ui/browser_commands.h"
 
+#include "brave/browser/new_tab/new_tab_shows_options.h"
 #include "brave/browser/profiles/profile_util.h"
 #include "chrome/browser/ui/browser_commands.h"
+
 #define ReloadBypassingCache ReloadBypassingCache_ChromiumImpl
+#define NewTab NewTab_ChromiumImpl_Unused
 #include "../../../../../chrome/browser/ui/browser_commands.cc"  // NOLINT
 #undef ReloadBypassingCache
+#undef NewTab
 
 namespace chrome {
 
+// Whole logic is copied from upstream chrome::NewTab().
```

Comment:
Maybe also note as part of this comment:
`// the only change is `new_tab_url` instead of `Gurl()`